pytest_factoryboy/fixture.py

```diff
@@ -118,41 +123,58 @@ def generate_fixturedefs(
         value = overrides.get(attr, value)
         attr_name = SEPARATOR.join((model_name, attr))
         yield (
+            attr_name,
             make_declaration_fixturedef(
                 attr_name=attr_name,
                 value=value,
                 factory_class=factory_class,
                 related=related,
-            )
+            ),
         )
 
     if factory_name not in caller_locals:
         yield (
-            FixtureDef(
+            factory_name,
+            create_fixture_with_related(
                 name=factory_name,
-                function_name="factory_fixture",
-                function_kwargs={"factory_class": factory_class},
-            )
+                function=functools.partial(factory_fixture, factory_class=factory_class),
+            ),
         )
 
     deps = get_deps(factory_class, model_name=model_name)
     yield (
-        FixtureDef(
+        model_name,
+        create_fixture_with_related(
             name=model_name,
-            function_name="model_fixture",
-            function_kwargs={"factory_name": factory_name},
-            deps=deps,
+            function=functools.partial(model_fixture, factory_name=factory_name),
+            fixtures=deps,
             related=related,
-        )
+        ),
     )
 
 
+def create_fixture_with_related(
+    name: str,
+    function: Callable[P, T],
+    fixtures: Collection[str] | None = None,
+    related: Collection[str] | None = None,
```

Comment:
Why `Collection` is used here instead of for instance `Sequence`?

Reply:
the difference is that Sequence implements Reversible, while Collection doesn't.

But we actually don't even need `fixtures` to be a Collection here, it can be an iterable, since we consume it once.
`related`, on the other hand, can't be an iterable.